### PR TITLE
[CL-46] Dialogs

### PR DESCRIPTION
**Pull request by Hinton**
## Type of change

```
- [ ] Bug fix
- [x] New feature development
- [ ] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other
```

## Objective

Renames Modals to Dialogs to better align with CDK naming convention. Also adds a new DialogService which is extends the CDK with some custom default settings.

Changed the max-widths of modals to width.

## Before you submit

```
- [ ] I have checked for **linting** errors (`npm run lint`) (required)
- [ ] I have added **unit tests** where it makes sense to do so (encouraged but not required)
- [ ] This change requires a **documentation update** (notify the documentation team)
- [ ] This change has particular **deployment requirements** (notify the DevOps team)
```

### CODE REVIEW

**Comment by Thomas-Avery** [comment]
@Hinton 

Looks like sizing of the dialog isn't working as expected. 

When specifying large `<bit-modal modalSize="large">`:

image

The styling of `tw-max-w-4xl` gets applied but the dialog doesn't increase in width.


It looks like the angular CDK .open() method controls sizing of the popup. 

We will have to rethink how to provide sizing options for consumers.

```
  openDialog() {
    this.dialogService.open(StoryModalContentComponent, {
      data: {
        animal: "panda",
      },
      height: '400px',
      width: '600px',
    });
  }
```

image

**Comment by Hinton** [comment]
I mean, technically it does exactly what it should do. Namely set the max width. It just doesn't have any content. If we want the modals to always have that size we will need to use widths and not max widths.

File: libs/components/src/dialog/dialgo-service.stories.ts [hunk 1]
```diff
@@ -0,0 +1,78 @@
+import { DialogModule, DialogRef, DIALOG_DATA } from "@angular/cdk/dialog";
```
Comment (vincentsalucci) [inline libs/components/src/dialog/dialgo-service.stories.ts:1]: File name typo! dialgo -> dialog

:D

Comment (Hinton) [reply libs/components/src/dialog/dialgo-service.stories.ts:1]: What no? I definitely mean Dial Go!

**Review by danielleflinn — APPROVED** [review]

**Review by vincentsalucci — APPROVED** [review]

**Pull request merged**